src/data/__tests__/index.test.ts

```diff
@@ -0,0 +1,80 @@
+import { UpsertCommand } from '../upsert';
+import { FetchCommand } from '../fetch';
+import { Index } from '../index';
+
+jest.mock('../upsert');
+jest.mock('../fetch');
+
+describe('Index', () => {
```

Comment:
All the actual functionality of the index actions is tested in more detail elsewhere. I added these tests to see whether the generic type param was resulting in the expected typescript errors. Expected errors are marked `// @ts-expect-error` and the compiler errors if no exception occurs.